src/init.cpp

```diff
@@ -1410,7 +1411,18 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     LogPrintf("Using at most %i connections (%i file descriptors available)\n", nMaxConnections, nFD);
     std::ostringstream strErrors;
 
-    InitSignatureCache();
+    // Initialize the validity caches. We currently have three:
+    // - Transparent signature validity.
+    // - Sapling bundle validity.
+    // - Orchard bundle validity.
+    // Assign half of the cap to transparent signatures, and split the rest
+    // between Sapling and Orchard bundles.
+    size_t nMaxCacheSize = GetArg("-maxsigcachesize", DEFAULT_MAX_SIG_CACHE_SIZE) * ((size_t) 1 << 20);
+    if (nMaxCacheSize <= 0) {
+        return InitError(strprintf(_("-maxsigcachesize must be at least 1")));
+    }
+    InitSignatureCache(nMaxCacheSize / 2);
+    bundlecache::init(nMaxCacheSize / 4);
```

Comment:
What happens if `nMaxCacheSize / 2` or `nMaxCacheSize / 4` becomes `0`? Then the `CuckooCache` instance later is initialized with `0` bytes, but the code previously demanded that `maxsigcachesize` be at least `1`. Seems like we might want to do `if (nMaxCacheSize < 8) {` now instead, and for good measure we could do `nMaxCacheSize >> 1` and `nMaxCacheSize >> 2` for clarity.

Reply:
`GetArg` returns an `int`, which is then multiplied by 2<sup>20</sup>, which is therefore the smallest positive value it can take (giving a bundle cache size of 2<sup>5</sup> bytes).

Reply:
Oh, right. I thought it was being kept around as a multiple of the number of megabytes but no, we rightshift when we want to recover that later for logs and stuff.

Reply:
You mean "giving a bundle cache size of 2<sup>18</sup> bytes".

Reply:
Indeed, I can definitely math.